includes/wp-admin/table/class-following.php

```diff
@@ -123,11 +124,34 @@ public function process_action() {
 				}
 
 				$profile = \sanitize_text_field( \wp_unslash( $_REQUEST['activitypub-profile'] ) );
-				if ( ! \is_email( \ltrim( $profile, '@' ) ) && empty( \wp_parse_url( $profile, PHP_URL_SCHEME ) ) ) {
+				$profile = \trim( $profile, '@' );
+
+				if ( \filter_var( $profile, FILTER_VALIDATE_EMAIL ) ) {
+					$remote_actor = Webfinger::resolve( $profile );
+					if ( ! \is_wp_error( $remote_actor ) ) {
+						$profile = object_to_uri( $remote_actor );
+					}
+				} elseif ( empty( \wp_parse_url( $profile, PHP_URL_SCHEME ) ) ) {
 					// Add scheme if missing.
 					$profile = \esc_url_raw( 'https://' . \ltrim( $profile, '/' ) );
 				}
 
+				// Check if user has blocked the account.
+				if ( \in_array( $profile, Moderation::get_user_blocks( $this->user_id )['actors'], true ) ) {
+					/* translators: %s: Account profile that could not be followed */
+					\add_settings_error( 'activitypub', 'followed', \sprintf( \__( 'Unable to follow account &#8220;%s&#8221;. The account is blocked.', 'activitypub' ), \esc_html( $profile ) ) );
+					$redirect_to = \add_query_arg( 'resource', $profile, $redirect_to );
+					break;
+				}
+
+				// Check if site has blocked the account.
+				if ( \in_array( $profile, Moderation::get_site_blocks()['actors'], true ) ) {
```

Comment:
we should have a dedicated function for that check, something like `is_actor_blocked( $uri )` to not have too much array comparisons in the places we want to have this check.